### PR TITLE
Added support for CentOS

### PR DESCRIPTION
**Pull request by hemanth-avs**
What this PR does: Support CentOS

**Which issue(s) this PR fixes** : Issue #73

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc @xunpan

**Special notes for your reviewer**:

/sig ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes. - No

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by gyliu513** [comment]
@hemanth-avs can you please sign the CLA first?

/ok-to-test

/cc @hchenxa

**Comment by gyliu513** [comment]
CI is failing due to 
```
# Create a dummy file for test only
echo 'clouds' > cmd/clusterctl/examples/ibmcloud/dummy-clouds-test.yaml
cmd/clusterctl/examples/ibmcloud/generate-yaml.sh -f dummy-clouds-test.yaml ubuntu dummy-make-auto-test
provider-os error: ubuntu is not one of the supported operating systems!
generate-yaml.sh - generates a provider-configs.yaml file
Usage:
generate-yaml.sh [options] <path/to/clouds.yaml> <provider os: [ubuntu/centos]> [output folder]
options:
-h, --help show brief help
-f, --force-overwrite if file to be generated already exists, force script to overwrite it
Makefile:127: recipe for target 'generate_yaml_test' failed
make: *** [generate_yaml_test] Error 1
```

**Comment by jichenjc** [comment]
@hemanth-avs that test was for a dummy test added to test
cmd/clusterctl/examples/ibmcloud/generate-yaml.sh

'provider-os error: ubuntu is not one of the supported operating systems!'

maybe you can manually do some test first locally

File: cmd/clusterctl/examples/ibmcloud/generate-yaml.sh [hunk 1]
```diff
@@ -7,15 +7,15 @@ print_help()
...
+  echo "$SCRIPT [options] <path/to/clouds.yaml> <provider os: [ubuntu/centos]> [output folder]"
   echo "options:"
   echo "-h, --help                    show brief help"
   echo "-f, --force-overwrite         if file to be generated already exists, force script to overwrite it"
   echo ""
 }
 
 # Supported Operating Systems
-declare -a arr=("ubuntu")
+declare -a arr=("ubuntu", "centos")
```
Comment (jichenjc) [inline cmd/clusterctl/examples/ibmcloud/generate-yaml.sh]: declare -a arr=("centos" "ubuntu" )

use this @hemanth-avs

**Pull request closed without merging**